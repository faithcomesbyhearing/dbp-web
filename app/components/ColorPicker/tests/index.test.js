import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Colors from '../../../../theme_config/javascriptColors';
import ColorPicker from '..';

const green = Colors.highlightGreen;
const yellow = Colors.highlightYellow;
const pink = Colors.highlightPink;
const purple = Colors.highlightPurple;
const blue = Colors.highlightBlue;

describe('ColorPicker tests', () => {
	let handlePickedColor;

	beforeEach(() => {
		handlePickedColor = jest.fn();
	});

	it('Should render and match snapshot', () => {
		const { container } = render(
			<ColorPicker handlePickedColor={handlePickedColor} />,
		);
		expect(container).toMatchSnapshot();
	});

	it('Should call the handler when the green option is clicked', () => {
		const { getByLabelText } = render(
			<ColorPicker handlePickedColor={handlePickedColor} />,
		);
		const greenButton = getByLabelText('green');

		fireEvent.click(greenButton);
		expect(handlePickedColor).toHaveBeenCalledWith({ color: green });
		expect(handlePickedColor).toHaveBeenCalledTimes(1);
	});

	it('Should call the handler when the yellow option is clicked', () => {
		const { getByLabelText } = render(
			<ColorPicker handlePickedColor={handlePickedColor} />,
		);
		const yellowButton = getByLabelText('yellow');

		fireEvent.click(yellowButton);
		expect(handlePickedColor).toHaveBeenCalledWith({ color: yellow });
		expect(handlePickedColor).toHaveBeenCalledTimes(1);
	});

	it('Should call the handler when the pink option is clicked', () => {
		const { getByLabelText } = render(
			<ColorPicker handlePickedColor={handlePickedColor} />,
		);
		const pinkButton = getByLabelText('pink');

		fireEvent.click(pinkButton);
		expect(handlePickedColor).toHaveBeenCalledWith({ color: pink });
		expect(handlePickedColor).toHaveBeenCalledTimes(1);
	});

	it('Should call the handler when the purple option is clicked', () => {
		const { getByLabelText } = render(
			<ColorPicker handlePickedColor={handlePickedColor} />,
		);
		const purpleButton = getByLabelText('purple');

		fireEvent.click(purpleButton);
		expect(handlePickedColor).toHaveBeenCalledWith({ color: purple });
		expect(handlePickedColor).toHaveBeenCalledTimes(1);
	});

	it('Should call the handler when the blue option is clicked', () => {
		const { getByLabelText } = render(
			<ColorPicker handlePickedColor={handlePickedColor} />,
		);
		const blueButton = getByLabelText('blue');

		fireEvent.click(blueButton);
		expect(handlePickedColor).toHaveBeenCalledWith({ color: blue });
		expect(handlePickedColor).toHaveBeenCalledTimes(1);
	});
});
