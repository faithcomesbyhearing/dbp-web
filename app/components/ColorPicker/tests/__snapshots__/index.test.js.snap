// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker tests Should render and match snapshot 1`] = `
<div>
  <div
    class="color-picker"
  >
    <span
      class="color-group"
    >
      <span
        aria-label="yellow"
        class="yellow"
        role="button"
        style="background-color: rgba(252, 230, 0, 0.5);"
        tabindex="-1"
      />
    </span>
    <span
      class="color-group"
    >
      <span
        aria-label="green"
        class="green"
        role="button"
        style="background-color: rgba(84, 185, 72, 0.5);"
        tabindex="-2"
      />
    </span>
    <span
      class="color-group"
    >
      <span
        aria-label="pink"
        class="pink"
        role="button"
        style="background-color: rgba(208, 105, 169, 0.5);"
        tabindex="-3"
      />
    </span>
    <span
      class="color-group"
    >
      <span
        aria-label="purple"
        class="purple"
        role="button"
        style="background-color: rgba(137, 103, 172, 0.5);"
        tabindex="-4"
      />
    </span>
    <span
      class="color-group"
    >
      <span
        aria-label="blue"
        class="blue"
        role="button"
        style="background-color: rgba(80, 165, 220, 0.5);"
        tabindex="-5"
      />
    </span>
  </div>
</div>
`;
